tools/Microsoft.TemplateSearch.TemplateDiscovery/NuGet/NugetPackProvider.cs

```diff
@@ -143,20 +139,28 @@ public async Task<IDownloadedPackInfo> DownloadPackageAsync(ITemplatePackageInfo
 
             try
             {
-                using Stream packageStream = File.Create(outputPackageFileNameFullPath);
-                if (await _downloadResource.CopyNupkgToStreamAsync(
-                    packinfo.Name,
-                    new NuGetVersion(packinfo.Version!),
-                    packageStream,
-                    _cacheContext,
-                    NullLogger.Instance,
-                    token).ConfigureAwait(false))
+                if (File.Exists(outputPackageFileNameFullPath))
                 {
+                    using Stream existingPackageStream = File.OpenRead(outputPackageFileNameFullPath);
```

Comment:
What's the purpose of just opening the file. Doesn't seem like you're doing anything with the stream.

Reply:
good eye, I was on autopilot through this part. we should be able to skip this entirely.